src/Common/MessageBroker/Event/MessageReturned.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gaming\Common\MessageBroker\Event;
+
+use Gaming\Common\MessageBroker\Message;
+
+final class MessageReturned
+{
+    /**
+     * @param array<string, string> $metadata
+     */
+    public function __construct(
+        public readonly Message $message,
+        public readonly array $metadata
```

Comment:
- Add `public readonly string $cause` so integrations can give a hint to a listener. In case of RabbitMQ it could be `basic.nack` or `basic.return`.
- Add `public function retry(): void`.
- Maybe add another property that tells how many times the message got retried.
- It should be possible to stop this event. If a listener decides to retry 5 times, it could stop the propagation 5 times, but then pass the responsibility to the listener, which could, e.g., store the returned message in another storage or log critical and throw an exception.

Reply:
The first is done. The last 3 can be an enhancement in future iterations.